hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java

```diff
@@ -195,43 +197,30 @@ public TransactionInfo runAllChecks(
         return txInfo;
     }
 
-    private void verifyPayerSignature(@NonNull final TransactionInfo txInfo, @NonNull final Key payerKey)
+    private void verifyPayerSignature(
+            @NonNull final TransactionInfo txInfo,
+            @NonNull final Account payer,
+            @NonNull final Key payerKey,
+            @NonNull final Configuration configuration)
             throws PreCheckException {
+        final var hederaConfig = configuration.getConfigData(HederaConfig.class);
+        final var sigPairs = txInfo.signatureMap().sigPairOrElse(List.of());
+
         // Expand the signatures
         final var expandedSigs = new HashSet<ExpandedSignaturePair>();
-        signatureExpander.expand(payerKey, txInfo.signatureMap().sigPairOrThrow(), expandedSigs);
+        signatureExpander.expand(sigPairs, expandedSigs);
+        if (!isHollow(payer)) {
+            signatureExpander.expand(payerKey, sigPairs, expandedSigs);
+        }
```

Comment:
I'm not sure about this part either, as to why we check for hollow. In the other file we also check for null payer. It just isn't clear to me what the expectation is

Reply:
Same here. I will go thoroughly through this next week.